### Задача 1 ###
Вы провели эксперимент c упрощением формы заказа в магазине Утконос и получили результаты по метрике конверсий в покупку. Выберите метод оценки и оцените есть ли статистически значимые различия между конверсиями в двух группах при alpha = 5%. Дайте краткие рекомендации команде. Результаты:
- Число юзеров в группах , которые заходили на сайт в период эксперимента: n1 = 15550 и n2 = 15550 .
- Число юзеров в группах , которые совершили хотя бы одну покупку за период эксперимента: n1 = 164 и n2 = 228
- Конверсии : conv1 = 1.05% conv2 = 1.47%.


In [ ]:
from scipy.stats import norm
import numpy as np

users1 = 15550
users2 = 15550
buyers1 = 164
buyers2 = 228
trials = [n1, n2]

# Конверсии в группах
conv1 = buyers1 / users1
conv2 = buyers2 / users2
conv1*100, conv2*100

(1.0546623794212218, 1.4662379421221865)

In [ ]:
# Объединенная конверсия
p = (buyers1 + buyers2) / (users1 + users2)
p*100

1.2604501607717042

In [ ]:
# z-статистика
z_stat = (conv1 - conv2) / np.sqrt(p * (1 - p) * (1/users1 + 1/users2))
z_stat

-3.253054722696254

In [ ]:
# p-value (двусторонний тест)
p_value = norm.sf(abs(z_stat))*2
p_value

0.001141715077067056

На основе z-статистики: -3.2531 и p-value: 0.0011, можно сделать следующие выводы:

- z-статистика отрицательна, что означает, что конверсия в первой группе (buyers1 / users1) меньше, чем конверсия во второй группе (buyers2 / users2).
- p-value меньше общепринятого уровня значимости 0.05, что означает, что разница между конверсиями в двух группах статистически значима.

Таким образом, можно сделать вывод о том, что конверсия во второй группе статистически значимо выше, чем конверсия в первой группе.

На основе статистически значимой разницы в конверсиях между двумя версиями сайта магазина Утконос (как указано в вашем первом сообщении), можно предложить следующие рекомендации:

Рассмотреть возможность использования новой версии сайта (версия B) как основной, так как она показывает более высокую конверсию по сравнению с текущей версией (версия A).

Провести дополнительные исследования для выявления причин более высокой конверсии в версии B. Это может помочь определить, какие изменения в дизайне или функциональности сайта наиболее положительно влияют на конверсию.

Продолжить проведение A/B-тестирования для оптимизации дизайна и функциональности сайта с целью увеличения конверсии.

Эти рекомендации могут помочь магазину Утконос улучшить свой сайт и увеличить продажи


### Задача 2 ### 
Сравниваем метрику конверсия в покупку. Размер выборки - 10000 элементов в каждой группе . Какой статистический критерий тут лучше всего подойдёт и почему?

Для сравнения конверсии в покупку между двумя группами можно использовать статистический критерий под названием двухвыборочный тест пропорций или Z-тест1. Этот тест используется для сравнения пропорций двух групп и определения того, являются ли они статистически значимо различными.

Z-тест является общепринятым методом для сравнения пропорций двух групп. Он используется в тех случаях, когда размер выборки достаточно большой и данные распределены нормально. В наешь случае размер выборки составляет 10000 элементов в каждой группе, что является достаточным для использования Z-теста.